### PR TITLE
restructuring suggestions

### PR DESCRIPTION
**Pull request by Necktschnagge**
I changed some things, made some experiments.
Let's discuss this and apply some further enhancements before merging into your branch.

### CODE REVIEW

**Comment by r0ost3r** [comment]
on/ off switch for solver @Necktschnagge 
https://stackoverflow.com/questions/14780517/toggle-switch-in-qt
no need to place solver in status bar

**Comment by r0ost3r** [comment]
may be we should use `QStatusBar::insertWidget` or `QStatusBar::addWidget` for _Color_ instead of **Permanent**Widget @Necktschnagge

**Comment by Necktschnagge** [comment]
Color ball looks nice in code @r0ost3r.
Can you add it to the status bar?

File: src/mainwindow.h [hunk 1]
```diff
@@ -24,14 +25,26 @@ class MainWindow : public QMainWindow
 {
 	Q_OBJECT
 
+private:
+	struct StatusbarItems {
+		QLabel* stepsKey;
+		QLabel* stepsValue;
+
+		// add more labels: "BoardId:" "702563:378 // 3:0:2:3:5:2:7 " hint: redPlanetQuadrant, .., .., .., permutation, board rotation, target cell
```
Comment (Necktschnagge) [inline src/mainwindow.h:33]: Can you add the requested additional QLabels here,
you may also add your color ball here.
I'd also be fine with just a Label with empty space and the correct color.
Feel free to find your favourite solution.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -245,5 +236,43 @@ void MainWindow::on_actionWEST_triggered()
...
 void MainWindow::setNumberOfSteps(QString& c)
 {
-    countNumberOfSteps->setText(c);
+	statusbarItems.stepsValue->setText(c);
+}
+
+void MainWindow::StatusbarItems::init(QStatusBar* statusbar) {
+	stepsKey = new QLabel(statusbar); // parent takes ownership
+	stepsValue = new QLabel(statusbar); // parent takes ownership
+
```
Comment (Necktschnagge) [inline src/mainwindow.cpp:245]: Here you'd have to add the initialisation code for additional labels or other widgets inside status bar.

**Review by Necktschnagge — COMMENTED** [review]
Things you can implement @r0ost3r

**Comment by Necktschnagge** [comment]
> on/ off switch for solver @Necktschnagge https://stackoverflow.com/questions/14780517/toggle-switch-in-qt no need to place solver in status bar

I don't want a toggle switch. Furthermore, solver will have more modes than just ON and OFF.

**Comment by Necktschnagge** [comment]
> may be we should use `QStatusBar::insertWidget` or `QStatusBar::addWidget` for _Color_ instead of **Permanent**Widget @Necktschnagge

No, let's stick to addPermanentWidget.
The other methods are intended to add labels that can be obscured by temporary messages.

That's not so nice.
Once we have too many labels we can start thinking about which of them are less important than others.

**Comment by Necktschnagge** [comment]
Let's code the rest together @r0ost3r

File: src/mainwindow.h (hunk 1 above)
Comment (Necktschnagge) [reply src/mainwindow.h:33]: @r0ost3r Can you add the labels as requested in these comments?

**Pull request merged**